### PR TITLE
`Programming exercise`: Enable viewing template and solution repositories

### PR DESCRIPTION
**Pull request by terlan98**
Adds functionality to view the solution and template repositories while assessing a programming submission. To avoid confusion, the app does not let the user add/edit feedback while viewing a non-student repository.

Demo:

https://github.com/ls1intum/Themis/assets/22798314/7989d381-d865-4b41-a5c3-d6305459ba55

### CODE REVIEW

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
Tested locally. During assessment, switching between the different repos works as expected.
However, in the read-only mode, there seems to be an issue:

https://github.com/ls1intum/Themis/assets/9954674/a7d497a8-8c81-45e8-a34e-20a160349d5d

**Comment by terlan98** [comment]
Thanks for the review! ~~This Artemis PR (https://github.com/ls1intum/Artemis/pull/6810) will fix the issue.~~

**Comment by terlan98** [comment]
The issue is now fixed 🙂

File: Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+import DesignLibrary
 
 struct FiletreeSidebarView: View {
     @ObservedObject var cvm: CodeEditorViewModel
     @ObservedObject var assessmentVM: AssessmentViewModel
+    @Binding var repositorySelection: RepositoryType
     
     var body: some View {
         VStack(alignment: .leading) {
+            Picker(selection: $repositorySelection) {
```
Comment (maximiliansoelch) [inline Themis/Views/Assessment/FiletreeSidebar/FiletreeSidebarView.swift:12]: Not sure yet, if this is the optimal position for the picker. We should talk about some alternative placements

**Review by maximiliansoelch — APPROVED** [review]
Code looks good. Also tested locally and viewing the template/solution repos works as expected 👍

**Pull request merged**